solidity/contracts/PausableReentrancyGuard.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+// adapted from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+abstract contract PausableReentrancyGuardUpgradeable is Initializable {
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = 2;
+    uint256 private constant _PAUSED = 3;
```

Comment:
I wonder if it might be more readable to treat the pasued/unpaused entered/not entered as two separate booleans? You could pack them into the two low order bits of the uint256 and the bitmasks 0x1 and 0x2 to read them

Reply:
imo introducing bitpacking/unpacking to the OZ logic is not worth it